File: Arrays and Hashing/238-Product-of-Array-Except-Self/238.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    const res = new Array(nums.length).fill(1);
    for(let i = 1; i < nums.length; i++){
        res[i] = res[i - 1] * nums[i - 1];
    }
    let postfix = 1;
    for(let i = nums.length - 1; i >= 0; i--){
        res[i] *= postfix;
        postfix *= nums[i];
    }
    return res;
};

File: Arrays and Hashing/238-Product-of-Array-Except-Self/238.cs
public class Solution {
    public int[] ProductExceptSelf(int[] nums) {
        int[] res = new int[nums.Length];
        Array.Fill(res, 1);
        for(int i = 1; i < nums.Length; i++){
            res[i] = res[i - 1] * nums[i - 1];
        }
        int postfix = 1;
        for(int i = nums.Length - 1; i >= 0; i--){
            res[i] *= postfix;
            postfix *= nums[i];
        }
        return res;
    }
}